packages/http-framework/src/lib/interactions/structures/Interaction.ts

```diff
@@ -263,9 +263,12 @@ export class Interaction<I extends BaseInteractionType = BaseInteractionType> {
 		return resultFromDiscord(container.rest.get(Routes.guild(this.guildId)) as Promise<RESTGetAPIGuildResult>);
 	}
 
-	protected async _sendReply(data: NonNullObject) {
-		if (this[Reply].sent) throw new Error('Cannot send response, the request has already been replied.');
-		await this[Reply].status(HttpCodes.OK).type('application/json').send(data);
+	protected _sendReply(data: NonNullObject) {
+		const response = this[Response];
+		if (response.closed) throw new Error('Cannot send response, the request has already been replied.');
+
+		response.statusCode = HttpCodes.OK;
+		return new Promise<void>((resolve) => response.end(JSON.stringify(data), resolve));
```

Comment:
What's the resolve here for? Also does it silence errors?

Reply:
It's called when the data has been fully sent, see [docs](https://nodejs.org/dist/latest-v18.x/docs/api/stream.html#writableendchunk-encoding-callback).